plugin/plugin.go

```diff
@@ -120,6 +140,17 @@ func (p *Plugin) Init(ctx context.Context, spec []byte, options NewClientOptions
 	}
 	defer p.mu.Unlock()
 	var err error
+
+	if p.schemaValidator != nil {
```

Comment:
Do we want to validate the top level schema too via JSON schema? Otherwise the validation error messages might be confusing (some in the JSON schema validator format and others in the JSON decoder format). We'll probably need to do this in the CLI

Reply:
Also maybe you can share how a validation error looks like from the console